go-controller/pkg/ovn/obj_retry.go

```diff
@@ -24,6 +40,45 @@ type retryObjEntry struct {
 	ignore bool
 }
 
+type retryObjs struct {
+	retryMutex sync.Mutex
+	// cache to hold object needs retry to successfully complete processing
+	entries map[string]*retryObjEntry
+	// resource type for these objects
+	oType reflect.Type
+	// channel to indicate we need to retry objs immediately
+	retryChan chan struct{}
+	// namespace filter fed to the handler for this resource type
+	namespaceForFilteredHandler string
+	// label selector fed to the handler for this resource type
+	labelSelectorForFilteredHandler labels.Selector
```

Comment:
are those two fields used only by pod ? or there are other resources that can use them ? not sure if extra param can be structure where those are there or we need them here wdyt ?

Reply:
These are the filters that we use when creating/adding a handler. They are empty for nodes, pods and network policies, but they are definitely in use for the six dynamic handlers that are created when adding a new network policy. All the existing functions that add a handler for a given resource type (https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L380-L483) already call `addHandler` with, among others, these two parameters: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L330-L374 
This said, I'm open of course to any renaming and refactoring option :)